gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/GradleCacheExplorer.java

```diff
@@ -31,90 +26,71 @@
 import java.util.HashSet;
 import java.util.Optional;
 import java.util.Set;
-import java.util.concurrent.Callable;
-import java.util.concurrent.Future;
-import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
+import org.apache.commons.lang3.time.StopWatch;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class GradleCacheExplorer {
 
     private static final Logger log = LoggerFactory.getLogger(GradleCacheExplorer.class);
     private static final String GRADLE_CACHE_PATH = System.getProperty("user.home") + "/.gradle/caches/modules-2/";
-    private final Cache<String, Set<String>> cache =
-            Caffeine.newBuilder().expireAfterWrite(10, TimeUnit.MINUTES).build();
+    private final AtomicReference<Set<String>> cache = new AtomicReference<>(Collections.emptySet());
 
-    public final void invalidateCache() {
-        cache.invalidateAll();
+    GradleCacheExplorer() {
+        loadCache();
+    }
+
+    public final void loadCache() {
+        cache.set(extractStrings());
     }
 
     public final Set<String> getCompletions(Set<String> repoUrls, DependencyGroup input) {
+        StopWatch stopWatch = new StopWatch();
+        stopWatch.start();
+
         String parsedInput = String.join(".", input.parts());
-        ProgressIndicator indicator = ProgressManager.getInstance().getProgressIndicator();
-        if (indicator == null) {
-            return Collections.emptySet();
+
+        Set<String> results = cache.get().stream()
+                .map(url -> extractGroupAndArtifactFromUrl(repoUrls, url))
+                .flatMap(Optional::stream)
+                .collect(Collectors.toSet());
+
+        if (parsedInput.isEmpty()) {
+            return results;
         }
 
-        try {
-            Callable<Set<String>> task = () -> extractStrings(repoUrls, indicator);
-            Future<Set<String>> future = ApplicationManager.getApplication().executeOnPooledThread(task);
-            Set<String> results =
-                    com.intellij.openapi.application.ex.ApplicationUtil.runWithCheckCanceled(future::get, indicator);
+        stopWatch.stop();
+        log.debug("Cache parsing time: {} ms", stopWatch.getTime());
```

Comment:
Can we call this "completion matching time" or something? "Cache parsing time" sounds like what `extractStrings()` is doing.